s3fs/core.py

```diff
@@ -142,7 +169,10 @@ def _ls(self, path, refresh=False):
                     f['Size'] = 0
                     del f['Name']
             else:
-                files = self.s3.list_objects(Bucket=bucket).get('Contents', [])
+                try:
+                    files = self.s3.list_objects(Bucket=bucket).get('Contents', [])
+                except ClientError:
+                    files = []
```

Comment:
Looking around on SO it appears that `ClientError` is raised on missing bucket?  This seems to be verified when using locally:

```python
In [1]: from distributed.s3fs import S3FileSystem
s3 
In [2]: s3 = S3FileSystem()

In [3]: _ = s3.s3.list_objects(Bucket='dask-data')

In [4]: _ = s3.s3.list_objects(Bucket='shffhhf')
---------------------------------------------------------------------------
ClientError                               Traceback (most recent call last)
```

Maybe raise a `FileNotFoundError` in the `except` branch?

This doesn't resolve the case when a bucket exists but no keys do.  Again, I'd be tempted to raise `FileNotFoundError` here if no matches, thoughts?